src/include/network/routines.rb

```diff
@@ -742,24 +756,62 @@ def SetIfaceDown(dev_name)
 
     # Tries to set all interfaces up
     #
-    # @return [boolean] false if some of interfaces cannot be set up
+    # @return [boolean] false if interfaces are empty or some
+    # interface cannot be bring up
     def SetAllIfacesUp
+      all_up = true
       interfaces = GetAllInterfaces()
 
       return false if interfaces.empty?
 
-      interfaces.all? { |i| SetIfaceUp(i) }
+      interfaces.each { |i| all_up = false if !SetIfaceUp(i) }
+
+      all_up
```

Comment:
As you commented only `SetAllLinksUp()` is currently in use. At least in yast2-network code and I also don't expect that someone is using `SetAllLinksDown()`. `SetAllIfacesUp` and `SetAllIfacesDown()` was introduced in the reverted commit and I fixed it basically based in CR and as they use the same code structure and also the main purpose of the PR was to revert the removement of `set link down`.